### PR TITLE
[PanelStack] merge some files

### PR DESCRIPTION
**Pull request by giladgray**
cc @kadhirvelm 

- merge `PanelHeader` into `PanelView`
- move panel props symbols to `panelProps.ts`
- ❗️ panel title accepts `ReactNode` instead of just string

### CODE REVIEW

File: packages/core/src/components/panel-stack/panelProps.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    openPanel<P>(component: React.ComponentType<P & IPanelProps>, props: P, options: IPanelOptions): void;
+}
+
+export interface IPanelOptions {
+    /**
+     * The title to be displayed above this panel. It is also used as the text
+     * of the back button for any panel opened by this panel.
+     */
+    title?: React.ReactNode;
```
Comment (giladgray) [inline packages/core/src/components/panel-stack/panelProps.ts]: ❗️ widened type here to support any JSX

File: packages/core/src/components/panel-stack/panelStack.tsx [hunk 2]
```diff
@@ -7,54 +7,34 @@ import { CSSTransition, TransitionGroup } from "react-transition-group";
...
-}
-
-export interface IPanelStackProps extends IProps {
-    /** The panel to show when initially rendering the component */
     initialPanel: IPanel;
+
     /**
-     * Callback invoked when the user presses the back button or a panel wants to close the visible panel.
+     * Callback invoked when the user presses the back button or the
+     * `closePanel()` injected prop method is invoked.
```
Comment (kadhirvelm) [inline packages/core/src/components/panel-stack/panelStack.tsx]: Small grammatical fix: "...back button or invokes the `closePanel()` injected prop method."

File: packages/core/src/components/panel-stack/panelView.tsx [hunk 3]
```diff
@@ -25,22 +27,35 @@ export interface IPanelViewProps {
...
-            <PanelHeader backTitle={this.props.previousPanel.title} onBackClick={this.props.panelProps.closePanel}>
-                {this.props.panel.title}
-            </PanelHeader>
+            <Button
+                className={Classes.PANEL_STACK_HEADER_BACK}
+                icon="chevron-left"
+                minimal={true}
+                small={true}
+                text={this.props.previousPanel.title}
+                onClick={this.props.panelProps.closePanel}
```
Comment (kadhirvelm) [inline packages/core/src/components/panel-stack/panelView.tsx]: Probably want to alphabetize the `onClick` here

File: packages/core/src/components/panel-stack/panelStack.tsx (hunk 2 above)
Comment (giladgray) [reply packages/core/src/components/panel-stack/panelStack.tsx]: but the user doesn't invoke the method

Comment (kadhirvelm) [reply packages/core/src/components/panel-stack/panelStack.tsx]: Yeah that's fair, I was trying to match the present simple tense from "user presses"

Comment (kadhirvelm) [reply packages/core/src/components/panel-stack/panelStack.tsx]: Actually I think it's mostly fine. Can you add an "either" before "the user" to more clearly separate the clauses?

**Pull request merged**